### PR TITLE
Add Dockerfile to build testiin docker image

### PR DESCRIPTION
**Pull request by ghoshbishakh**

### CODE REVIEW

File: tests/network-setups/indy/docker/Makefile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+iinimage:
+	docker build -t testiin .
+
+runiin:
```
Comment (VRamakrishna) [inline tests/network-setups/indy/docker/Makefile:4]: Just call this `start` instead of `runiin`.

File: tests/network-setups/indy/docker/Makefile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+iinimage:
```
Comment (VRamakrishna) [inline tests/network-setups/indy/docker/Makefile:1]: Just call this `image`.

File: tests/network-setups/indy/docker/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Modified indy network docker image
```
Comment (VRamakrishna) [inline tests/network-setups/indy/docker/README.md:1]: Capitalize "Indy".

File: tests/network-setups/indy/docker/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+Build iin docker image:
```
Comment (VRamakrishna) [inline tests/network-setups/indy/docker/README.md:3]: Capitalize IIN, here and elsewhere.

File: tests/network-setups/indy/docker/README.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Modified indy network docker image
+
```
Comment (VRamakrishna) [inline tests/network-setups/indy/docker/README.md:2]: Provide some more context here about what we are starting. Say that this is a sample or reference IIN, or decentralized identity registry, of the form that is specified in the RFCs (link to the appropriate RFC), built on Hyperledger Indy (link to that too).

File: tests/network-setups/indy/docker/run_indy_nodes.sh [hunk 6]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/src/indy_projects/ursa/target/release/
+generate_indy_pool_transactions --nodes 4 --clients 5 --nodeNum 1 2 3 4
```
Comment (VRamakrishna) [inline tests/network-setups/indy/docker/run_indy_nodes.sh:3]: With a compose file, we can use environment variables to configure the number of nodes and clients.

File: tests/network-setups/indy/docker/run_indy_nodes.sh [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+start_indy_node Node1 0.0.0.0 9701 0.0.0.0 9702 &
+
+start_indy_node Node2 0.0.0.0 9703 0.0.0.0 9704 &
+
+start_indy_node Node3 0.0.0.0 9705 0.0.0.0 9706 &
+
+start_indy_node Node4 0.0.0.0 9707 0.0.0.0 9708 &
```
Comment (VRamakrishna) [inline tests/network-setups/indy/docker/run_indy_nodes.sh:11]: Is it possible to run the above in a loop so we can start up an arbitrary number of nodes? Is there a limitation in Indy that we should be aware of that requires us to run exactly 4 nodes?

File: tests/network-setups/indy/docker/Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (ghoshbishakh) [reply tests/network-setups/indy/docker/Makefile]: Added stop.
What would clean do?

[the remark this answers is not in the document]
Comment (ghoshbishakh) [reply tests/network-setups/indy/docker/Makefile]: Delete if some files are generated in mounted volume. E.g. the genesis block.

**Comment by ghoshbishakh** [comment]
This PR resolves #191

**Comment by ghoshbishakh** [comment]
Manually set container name in dockerfile.
Mount genesis block directory.
Poll for a file.
docker cp

**Comment by ghoshbishakh** [comment]
Mention docker-compose version in readme.

**Comment by ghoshbishakh** [comment]
Updated file permissions from within docker.

File: tests/network-setups/indy/docker/Makefile [hunk 8]
```diff
@@ -0,0 +1,11 @@
+image:
+	docker build -t testiin .
+
+start:
+	./start_testiin.sh
+
+stop:
+	./stop_testiin.sh
+
+clean:
```
Comment (VRamakrishna) [inline tests/network-setups/indy/docker/Makefile:10]: Add `stop` to this as a target to complete before cleanup (i.e., after the colon). We shouldn't try to cleanup without stopping the network first.

Comment (VRamakrishna) [reply tests/network-setups/indy/docker/Makefile:10]: I'll add this in my next PR.

**Review by VRamakrishna — APPROVED** [review]

**Pull request merged**